### PR TITLE
Fixes minor issues found by static analyzer

### PR DESCRIPTION
**Pull request by qarmin**
Change some code with help static analysers

### CODE REVIEW

File: modules/assimp/editor_scene_importer_assimp.cpp [hunk 1]
```diff
@@ -854,7 +854,7 @@ Ref<Material> EditorSceneImporterAssimp::_generate_material_from_index(ImportSta
 			if (found) {
 				Ref<Texture> texture = _load_texture(state, path);
 
-				if (texture != NULL) {
+				if (texture != Ref<Texture>()) {
```
Comment (akien-mga) [inline modules/assimp/editor_scene_importer_assimp.cpp]: Should be `texture.is_valid()`.

File: modules/websocket/websocket_multiplayer_peer.cpp [hunk 2]
```diff
@@ -197,7 +197,7 @@ PoolVector<uint8_t> WebSocketMultiplayerPeer::_make_pkt(uint32_t p_type, int32_t
 	out.resize(PROTO_SIZE + p_data_size);
 
 	PoolVector<uint8_t>::Write w = out.write();
-	copymem(&w[0], &p_type, 1);
+	copymem(&w[0], &p_type, 4);
```
Comment (akien-mga) [inline modules/websocket/websocket_multiplayer_peer.cpp]: CC @Faless

**Comment by akien-mga** [comment]
Could you make the commit log more explicit? "Fixes minor issues found by static analyzer" would be better.

File: modules/websocket/websocket_multiplayer_peer.cpp (hunk 2 above)
Comment (Faless) [reply modules/websocket/websocket_multiplayer_peer.cpp]: This is wrong, the right fix is:
```diff
diff --git a/modules/websocket/websocket_multiplayer_peer.cpp b/modules/websocket/websocket_multiplayer_peer.cpp
index 23cbf916e..e24cb850e 100644
--- a/modules/websocket/websocket_multiplayer_peer.cpp
+++ b/modules/websocket/websocket_multiplayer_peer.cpp
@@ -191,7 +191,7 @@ void WebSocketMultiplayerPeer::_send_sys(Ref<WebSocketPeer> p_peer, uint8_t p_ty
        p_peer->put_packet(&(message.read()[0]), message.size());
 }
 
-PoolVector<uint8_t> WebSocketMultiplayerPeer::_make_pkt(uint32_t p_type, int32_t p_from, int32_t p_to, const uint8_t *p_data, uint32_t p_data_size) {
+PoolVector<uint8_t> WebSocketMultiplayerPeer::_make_pkt(uint8_t p_type, int32_t p_from, int32_t p_to, const uint8_t *p_data, uint32_t p_data_size) {
 
        PoolVector<uint8_t> out;
        out.resize(PROTO_SIZE + p_data_size);
diff --git a/modules/websocket/websocket_multiplayer_peer.h b/modules/websocket/websocket_multiplayer_peer.h
index 7fd97a659..e3ab0784a 100644
--- a/modules/websocket/websocket_multiplayer_peer.h
+++ b/modules/websocket/websocket_multiplayer_peer.h
@@ -41,7 +41,7 @@ class WebSocketMultiplayerPeer : public NetworkedMultiplayerPeer {
        GDCLASS(WebSocketMultiplayerPeer, NetworkedMultiplayerPeer);
 
 private:
-       PoolVector<uint8_t> _make_pkt(uint32_t p_type, int32_t p_from, int32_t p_to, const uint8_t *p_data, uint32_t p_data_size);
+       PoolVector<uint8_t> _make_pkt(uint8_t p_type, int32_t p_from, int32_t p_to, const uint8_t *p_data, uint32_t p_data_size);
        void _store_pkt(int32_t p_source, int32_t p_dest, const uint8_t *p_data, uint32_t p_data_size);
        Error _server_relay(int32_t p_from, int32_t p_to, const uint8_t *p_buffer, uint32_t p_buffer_size);
 
```

**Review by Faless — DISMISSED** [review]
See fix for WebSocketMultiplayer

**Comment by qarmin** [comment]
Added more fixes and also to .gitignore a CppCheck file.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**